rest/route/service.go

```diff
@@ -74,15 +76,15 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/cost/project/{project_id}/tasks").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeTaskCostByProjectRoute(sc))
 	app.AddRoute("/cost/version/{version_id}").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeCostByVersionHandler(sc))
 	app.AddRoute("/distros").Version(2).Get().Wrap(checkUser).RouteHandler(makeDistroRoute(sc))
-	app.AddRoute("/distros/{distro_id}").Version(2).Get().Wrap(superUser).RouteHandler(makeGetDistroByID(sc))
-	app.AddRoute("/distros/{distro_id}").Version(2).Patch().Wrap(superUser).RouteHandler(makePatchDistroByID(sc, settings))
-	app.AddRoute("/distros/{distro_id}").Version(2).Delete().Wrap(superUser).RouteHandler(makeDeleteDistroByID(sc))
-	app.AddRoute("/distros/{distro_id}").Version(2).Put().Wrap(superUser).RouteHandler(makePutDistro(sc, settings))
-	app.AddRoute("/distros/{distro_id}/setup").Version(2).Get().Wrap(superUser).RouteHandler(makeGetDistroSetup(sc))
-	app.AddRoute("/distros/{distro_id}/setup").Version(2).Patch().Wrap(superUser).RouteHandler(makeChangeDistroSetup(sc))
-	app.AddRoute("/distros/{distro_id}/teardown").Version(2).Get().Wrap(superUser).RouteHandler(makeGetDistroTeardown(sc))
-	app.AddRoute("/distros/{distro_id}/teardown").Version(2).Patch().Wrap(superUser).RouteHandler(makeChangeDistroTeardown(sc))
-	app.AddRoute("/distros/{distro_id}/execute").Version(2).Post().Wrap(superUser).RouteHandler(makeDistroExecute(sc, env))
+	app.AddRoute("/distros/{distro_id}").Version(2).Get().Wrap(superUser, viewDistroSettings).RouteHandler(makeGetDistroByID(sc))
+	app.AddRoute("/distros/{distro_id}").Version(2).Patch().Wrap(superUser, editDistroSettings).RouteHandler(makePatchDistroByID(sc, settings))
+	app.AddRoute("/distros/{distro_id}").Version(2).Delete().Wrap(superUser, editDistroSettings).RouteHandler(makeDeleteDistroByID(sc))
+	app.AddRoute("/distros/{distro_id}").Version(2).Put().Wrap(superUser, editDistroSettings).RouteHandler(makePutDistro(sc, settings))
```

Comment:
This route is a bit problematic since no one will ever have security to a new distro that's created. I think this should remain superuser only for now